src/main/webapp/resources/js/components/samples/components/SampleMetadata.jsx

```diff
@@ -9,26 +21,79 @@ import { Empty, List } from "antd";
  * @constructor
  */
 export function SampleMetadata({ metadata }) {
-  return Object.keys(metadata).length ? (
-    <List
-      itemLayout="horizontal"
-      dataSource={Object.keys(metadata).sort((a, b) => a.localeCompare(b))}
-      renderItem={(item) => (
-        <List.Item className="t-sample-details-metadata-item">
-          <List.Item.Meta
-            title={
-              <span className="t-sample-details-metadata__field">{item}</span>
-            }
-            description={
-              <span className="t-sample-details-metadata__entry">
-                {metadata[item].value}
-              </span>
-            }
+  const [visible, setVisible] = React.useState(false);
+  return (
+    <>
+      <Button onClick={() => setVisible(true)}>
+        {i18n("SampleMetadata.addNewMetadata")}
+      </Button>
+      <Modal
```

Comment:
Make this it's own component.  Use the sample coding style that I use for the root component of the sample details viewer to allow different types of buttons to open the modal.

Reply:
Sounds good! Thanks